From 723d41836db7669ab658d3e07c62fcbe17d7d7f4 Mon Sep 17 00:00:00 2001
From: zhengjunlong <zhengjunlong@huawei.com>
Date: Fri, 11 Oct 2024 17:08:35 +0800
Subject: [PATCH 01/17] include/msi: modify kabi size of msi_desc

hulk inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/IAW8JF

----------------------------------------------------

Change the size of the pre-embedded memory for msi_desc to 40 bytes.

Signed-off-by: Zheng Junlong <zhengjunlong@huawei.com>
---
 include/linux/msi.h | 11 ++++-------
 1 file changed, 4 insertions(+), 7 deletions(-)

diff --git a/include/linux/msi.h b/include/linux/msi.h
index 7354ffb14856..5fd8a6caae98 100644
--- a/include/linux/msi.h
+++ b/include/linux/msi.h
@@ -205,15 +205,12 @@ struct msi_desc {
 	union {
 		struct pci_msi_desc	pci;
 		struct msi_desc_data	data;
-		KABI_RESERVE(1)
-		KABI_RESERVE(2)
-		KABI_RESERVE(3)
-		KABI_RESERVE(4)
+		KABI_EXTEND_WITH_SIZE(KABI_RESERVE(1), 5)
 	};
+	KABI_RESERVE(2)
+	KABI_RESERVE(3)
+	KABI_RESERVE(4)
 	KABI_RESERVE(5)
-	KABI_RESERVE(6)
-	KABI_RESERVE(7)
-	KABI_RESERVE(8)
 };
 
 /*
-- 
2.25.1

